polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -788,6 +793,11 @@ export function TextField({
         verticalContentRef.current.contains(document.activeElement))
     );
   }
+
+  function isIconPrefix() {
+    if (!prefixRef.current) return false;
+    return Boolean(prefixRef.current.querySelector('svg'));
+  }
```

Comment:
I wonder if there is a way to do this without `querySelector` 🤔... I guess we don't know what is inside the `prefix`..... Is there anything we could do in CSS like check if there is a `prefix > svg` and adjust spacing?

Reply:
`:has` would be best for this because its definitely best to add the styles to the parent container over the `svg` element but its too [new](https://caniuse.com/css-has) for our supported browsers :( I can't really find an [older css equivalent to [`:has`](https://stackoverflow.com/a/2326600).

I'm pretty sure I can't adjust the spacing from the `svg` element or the `Icon` wrapper, but lmk if you think this is possible!

Reply:
`:has` is used 15 times in shopify/web 😬